packages/db/src/meta/pouch/adapters/base.ts

```diff
@@ -153,192 +146,115 @@ export abstract class Databases<C extends Collections> implements Workspace<C> {
-
-    return ({
-      [collectionName]: resourceInputIds.map(id => resourcesById[id])
-    } as unknown) as MutationPayload<C, N>;
-  }
+      .reduce(
+        (a, b) => ({ ...a, ...b }),
+        {} as { [id: string]: SavedRecord<I> }
+      );
 
-  public async update<M extends MutableCollectionName<C>>(
-    collectionName: M,
-    input: MutationInput<C, M>
-  ): Promise<MutationPayload<C, M>> {
-    await this.ready;
+    const savedRecordById: {
+      [id: string]: SavedRecord<I>;
+    } = (
+      await Promise.all(
+        Object.entries(recordsById).map(async ([_id, record]) => {
+          const existingSavedRecord = existingSavedRecordById[_id];
 
-    const log = debug.extend(`${collectionName}:update`);
-    log("Updating...");
+          if (existingSavedRecord && !overwrite) {
+            return existingSavedRecord;
+          }
 
-    const resourceInputIds = input[collectionName].map(
-      resourceInput => this.generateId<M>(collectionName, resourceInput) || ""
-    );
+          const { _rev = undefined } = existingSavedRecord || {};
 
-    const resourceInputById = input[collectionName]
-      .map((resourceInput, index) => ({
-        [resourceInputIds[index]]: resourceInput
-      }))
-      .reduce((a, b) => ({ ...a, ...b }), {});
-
-    const resources = await Promise.all(
-      Object.entries(resourceInputById)
-        .filter(([id]) => id)
-        .map(async ([id, resourceInput]) => {
-          // check for existing
-          const resource = await this.get(collectionName, id);
-          const { _rev = undefined } = resource ? resource : {};
-
-          await this.collections[collectionName].put({
-            ...resourceInput,
-            _rev,
-            _id: id
+          const { rev } = await this.collections[collectionName].put({
+            ...record,
+            _rev
           });
 
           return {
-            ...resourceInput,
-            id
-          } as SavedInput<C, M>;
+            ...record,
+            _rev: rev
+          } as SavedRecord<I>;
         })
-    );
-
-    const resourcesById = resources
-      .map(resource => ({
-        [resource.id as string]: resource
-      }))
-      .reduce((a, b) => ({ ...a, ...b }), {});
+      )
+    )
+      .map(savedRecord => ({ [savedRecord._id]: savedRecord }))
+      .reduce(
+        (a, b) => ({ ...a, ...b }),
+        {} as { [id: string]: SavedRecord<I> }
+      );
 
-    log(
-      "Updated ids: %o",
-      resources.map(({ id }) => id)
-    );
-    return ({
-      [collectionName]: resourceInputIds.map(id => resourcesById[id])
-    } as unknown) as MutationPayload<C, M>;
+    return records.map(record => record && savedRecordById[record._id]);
   }
 
-  public async remove<M extends MutableCollectionName<C>>(
-    collectionName: M,
-    input: MutationInput<C, M>
-  ): Promise<void> {
+  public async forget<N extends CollectionName<C>, T>(
+    collectionName: N,
+    references: (RecordReference<T> | undefined)[]
+  ) {
```

Comment:
Did you mean to drop the return type annotation here?

Reply:
I did think to myself, "ah yes, TS knows the return type from the interface" and proceeded to be lazy and not add it.

But this sorta thing is generally better listed explicitly :) I'll add them all.